Lib/test/test_regrtest.py

```diff
@@ -1452,7 +1452,7 @@ def test_fail_always(self):
         output = self.run_tests("--rerun", testname, exitcode=EXITCODE_BAD_TEST)
         self.check_executed_tests(output, [testname],
                                   rerun=Rerun(testname,
-                                              "test_fail_always",
+                                              "test_regrtest_noop\d+.Tests.test_fail_always",
```

Comment:
put the `r"` prefix on all of these containing a `\d` as well.  technically the `.` should also be replaced with a `\.` or `[.]` if matching on the actual . character itself is important.

(not a huge deal about the .s, it makes the test more lenient but it works and being ultra specific about the character wasn't the intent, it's just good regex practice.)

Reply:
I'm a bit surprised ruff didn't complain about the `r"` prefix. Anyway I fixed all of that.